modules/pubmaticBidAdapter.js

```diff
@@ -603,42 +612,85 @@ function _createImpressionObject(bid, conf) {
     bidfloorcur: bid.params.currency ? _parseSlotParam('currency', bid.params.currency) : DEFAULT_CURRENCY
   };
 
-  if (bid.params.hasOwnProperty('video')) {
-    var videoData = bid.params.video;
+  if (bid.hasOwnProperty('mediaTypes')) {
+    for (mediaTypes in bid.mediaTypes) {
+      switch (mediaTypes) {
+        case BANNER:
+          bannerObj = {};
+          sizes = bid.mediaTypes.banner.sizes;
+          if (sizes !== undefined && utils.isArray(sizes)) {
+            if (!bid.params.width && !bid.params.height) {
+              bannerObj.w = sizes[0][0];
+              bannerObj.h = sizes[0][1];
+              sizes = sizes.splice(1, sizes.length - 1);
+            } else {
+              bannerObj.w = bid.params.width;
+              bannerObj.h = bid.params.height;
+            }
+            if (sizes.length > 0) {
+              format = [];
+              sizes.forEach(function (size) {
+                format.push({ w: size[0], h: size[1] });
+              });
+              bannerObj.format = format;
+            }
+          } else {
+            utils.logWarn(BIDDER_CODE + 'Error: mediaTypes.banner.size missing for adunit: ' + bid.params.adUnit + '. Ignoring the adunit.');
+          }
+          bannerObj.pos = 0;
+          bannerObj.topframe = utils.inIframe() ? 0 : 1;
+          impObj.banner = bannerObj;
 
-    for (var key in VIDEO_CUSTOM_PARAMS) {
-      if (videoData.hasOwnProperty(key)) {
-        videoObj[key] = _checkParamDataType(key, videoData[key], VIDEO_CUSTOM_PARAMS[key])
-      }
-    }
-    // read playersize and assign to h and w.
-    if (utils.isArray(bid.mediaTypes.video.playerSize[0])) {
-      videoObj.w = bid.mediaTypes.video.playerSize[0][0];
-      videoObj.h = bid.mediaTypes.video.playerSize[0][1];
-    } else if (utils.isNumber(bid.mediaTypes.video.playerSize[0])) {
-      videoObj.w = bid.mediaTypes.video.playerSize[0];
-      videoObj.h = bid.mediaTypes.video.playerSize[1];
-    }
-    if (bid.params.video.hasOwnProperty('skippable')) {
-      videoObj.ext = {
-        'video_skippable': bid.params.video.skippable ? 1 : 0
+          break;
+        case NATIVE:
+          nativeObj['request'] = JSON.stringify(_createNativeRequest(bid.nativeParams));
+          if (!isInvalidNativeRequest) {
+            impObj.native = nativeObj;
+          } else {
+            utils.logWarn(BIDDER_CODE + 'Error: Error in Native adunit ' + bid.params.adUnit + '. Ignoring the adunit. Refer to ' + PREBID_NATIVE_HELP_LINK + ' for more details.');
+          }
+
+          break;
+        case VIDEO:
+          var videoData = bid.params.video;
```

Comment:
can we move following video specific code in a separate function like _createNativeRequest?
smaller functions make code easy to understand and maintain.